### ISSUE TITLE
Plausible UI performs a full scan of events table

### ISSUE THREAD

**Issue by Tristan971** [plausible/analytics#3329]
### Past Issues Searched

- [X] I have searched open and closed issues to make sure that the bug has not yet been reported

### Issue is a Bug Report

- [X] This is a bug report and not a feature request, nor asking for self-hosted support

### Using official Plausible Cloud hosting or self-hosting?

Self-hosting

### Describe the bug

As of Plausible 1.5 (and as follow-up to https://github.com/plausible/analytics/discussions/1565), Plausible's UI becomes unusable past a certain amount of historical events, due to one single query being completely unbound, causing a complete table scan across all partitions of plausible.events.

Looking at our query_log table, I see exactly 1 slow query (finishing in more than 3 seconds), the following one with a total of 72 seconds: `SELECT min(e0."timestamp") FROM "events" AS e0 WHERE (e0."domain" = 'mangadex.org')`

Here is the complete query information as seen by Clickhouse:
```
Query id: c185256a-3fd1-4426-8d06-c35db4aa055c

Row 1:
──────
type:                                  QueryFinish
event_date:                            2023-09-06
event_time:                            2023-09-06 03:25:08
event_time_microseconds:               2023-09-06 03:25:08.802127
query_start_time:                      2023-09-06 03:23:56
query_start_time_microseconds:         2023-09-06 03:23:56.924988
query_duration_ms:                     71877
read_rows:                             6505145787
read_bytes:                            162630340781
written_rows:                          0
written_bytes:                         0
result_rows:                           1
result_bytes:                          256
memory_usage:                          54458523
current_database:                      plausible
query:                                 SELECT min(e0."timestamp") FROM "events" AS e0 WHERE (e0."domain" = 'mangadex.org')
 FORMAT JSONCompact
formatted_query:
normalized_query_hash:                 6516661565783477090
query_kind:                            Select
databases:                             ['plausible']
tables:                                ['plausible.events']
columns:                               ['plausible.events.domain','plausible.events.timestamp']
partitions:                            ['plausible.events.202106','plausible.events.202107','plausible.events.202108','plausible.events.202109','plausible.events.202110','plausible.events.202111','plausible.events.202112','plausible.events.202201','plausible.events.202202','plausible.events.202203','plausible.events.202204','plausible.events.202205','plausible.events.202206','plausible.events.202207','plausible.events.202208','plausible.events.202209','plausible.events.202210','plausible.events.202211','plausible.events.202212','plausible.events.202301','plausible.events.202302','plausible.events.202303','plausible.events.202304','plausible.events.202305','plausible.events.202306','plausible.events.202307','plausible.events.202308','plausible.events.202309']
projections:                           []
views:                                 []
exception_code:                        0
exception:
stack_trace:
is_initial_query:                      1
user:                                  default
query_id:                              940aacdf-5fad-4ab4-9214-83d30943b659
address:                               ::ffff:10.233.87.240
port:                                  60682
initial_user:                          default
initial_query_id:                      940aacdf-5fad-4ab4-9214-83d30943b659
initial_address:                       ::ffff:10.233.87.240
initial_port:                          60682
initial_query_start_time:              2023-09-06 03:23:56
initial_query_start_time_microseconds: 2023-09-06 03:23:56.924988
interface:                             2
is_secure:                             0
os_user:
client_hostname:
client_name:
client_revision:                       0
client_version_major:                  0
client_version_minor:                  0
client_version_patch:                  0
http_method:                           2
http_user_agent:                       hackney/1.18.1
http_referer:
forwarded_for:
quota_key:
distributed_depth:                     0
revision:                              54474
log_comment:
thread_ids:                            [586,460,291,325,350,316,292,298,446,303,294,308,469,456,352,546,411,340,338,545,2291,330,524,559,347,356,306,403,3200,290,568,471,374,3187,342,536,2275,404,317,366,3179,463,2337,410,409,368,465,369,563,466,406,309,2346,464,561,3180,548,358,2339,593,371,565,2267,588,289,580,2355,375,472,569,355,396,590,346,353,3166,361,412,296,417,514,373,420,2325,339,2279,328]
ProfileEvents:                         {'Query':1,'SelectQuery':1,'QueriesWithSubqueries':1,'SelectQueriesWithSubqueries':1,'FileOpen':213,'ReadBufferFromFileDescriptorReadBytes':26376613451,'ReadCompressedBytes':26358107028,'CompressedReadBufferBlocks':1184983,'CompressedReadBufferBytes':110613580386,'OpenedFileCacheHits':175,'OpenedFileCacheMisses':213,'IOBufferAllocs':787,'IOBufferAllocBytes':62686498,'ArenaAllocChunks':16,'ArenaAllocBytes':65536,'FunctionExecute':165478,'MarkCacheHits':388,'CreatedReadBufferOrdinary':388,'DiskReadElapsedMicroseconds':135186316,'SelectedParts':124,'SelectedRanges':136,'SelectedMarks':795067,'SelectedRows':6505145787,'SelectedBytes':162630340781,'WaitMarksLoadMicroseconds':8457,'ContextLock':433,'RWLockAcquiredReadLocks':1,'RealTimeMicroseconds':1440039972,'UserTimeMicroseconds':288258326,'SystemTimeMicroseconds':66447621,'SoftPageFaults':259033,'OSIOWaitMicroseconds':44460000,'OSCPUWaitMicroseconds':546786410,'OSCPUVirtualTimeMicroseconds':354717353,'OSReadBytes':26053709824,'OSReadChars':26511745062,'OSWriteChars':423160,'QueryProfilerRuns':1755,'ThreadPoolReaderPageCacheMiss':231221,'ThreadPoolReaderPageCacheMissBytes':26376613451,'ThreadPoolReaderPageCacheMissElapsedMicroseconds':135186316,'AsynchronousReadWaitMicroseconds':607820605}
Settings:                              {}
used_aggregate_functions:              ['max','count','min']
used_aggregate_function_combinators:   []
used_database_engines:                 []
used_data_type_families:               []
used_dictionaries:                     []
used_formats:                          ['JSONCompact']
used_functions:                        ['equals','toYYYYMM']
used_storages:                         []
used_table_functions:                  []
used_row_policies:                     []
transaction_id:                        (0,0,'00000000-0000-0000-0000-000000000000')
asynchronous_read_counters:            {}

```

Important here are both of:
- read_rows: 6,505,145,787
- partitions: all of them

I'm not sure yet of why the partition key isn't being used by Clickhouse, but it isn't (Clickhouse 23.5, and no modifications done by us to the table), which causes:
```
clickhouse :) EXPLAIN SELECT min(e0."timestamp") FROM "events" AS e0 WHERE (e0."domain" = 'mangadex.org')

┌─explain─────────────────────────────────────┐
│ Expression ((Projection + Before ORDER BY)) │
│   Aggregating                               │
│     Expression (Before GROUP BY)            │
│       ReadFromMergeTree (plausible.events)  │
└─────────────────────────────────────────────┘

4 rows in set. Elapsed: 0.004 sec.
```

### Expected behavior

There are a couple of potential solutions.

1. The query is replaced with another one that gives the same information and avoids the fullscan

Maybe done in 2.0 with the new tables DDL, I'd have to look. That user_id in the order by certainly sounds like something that'd warrant the big data migration.

2. The query is executed asynchronously by Plausible with a higher timeout (since currently it doesn't wait for a response long enough to ever get it) and cached to some extent

This is better than nothing but still means potentially massive reads suddenly whenever the cache expires and someone tries to load the site's page on Plausible.

### Environment

```markdown
- Plausible: v1.5.1
- Clickhouse: 23.5
```

**Comment by ruslandoga** [comment]
👋 @Tristan971 

I think this query has been removed in v2 and instead the first event timestamp has been moved to sites table in PostgreSQL.

**Comment by Tristan971** [comment]
Well that is certainly a promising answer 🙏 

I was hesitant to propose that solution as I figured it might have been made dynamic to account for end-user (or customer pricing plan based) TTLs on the table. That said, I'd have argued that it is probably not a use-case worth causing such load on every site dashboard load, so I'd definitely agree with the team's approach here.

**Comment by ruslandoga** [comment]
Here's what I get when I run the query on my instance:

```sql
4dde5e6ce8fc :) select min(timestamp) from events_v2 where site_id = 1 and timestamp > '2022-01-01';

-- Query id: 8312200e-ae92-40ee-9bd3-ea894dbcf360

-- ┌──────min(timestamp)─┐
-- │ 2022-08-10 15:43:41 │
-- └─────────────────────┘

-- 1 row in set. Elapsed: 0.316 sec. Processed 597.61 thousand rows, 7.17 MB (1.89 million rows/s., 22.71 MB/s.)
```

**Comment by Tristan971** [comment]
Well... if the query is never called anymore that's fine. So I suppose I'll be hoping that your guess that it was moved to Postgres since is right 😅

**Comment by ruslandoga** [comment]
Yes, seems like this query is only used to populate `sites.stats_start_date` in PostgreSQL and afterwards it's read from there.

**Comment by Tristan971** [comment]
Well, I can confirm that 2.0 doesn't trigger this issue anymore, for anyone else looking at it.

Assuming you're not planning to try and fix it in 1.5, I'll just close the issue. But feel free to reopen if you do.
